### DISCUSSION TITLE
Multiple builds with different environments variables but not as a matrix

### DISCUSSION THREAD

**Discussion by zopyx** [community/community#25542] [Actions]
<p>I have this Travis configuration for a Python/Plone project:</p>
<p><a href="https://github.com/zopyx/xmldirector.connector/blob/master/.travis.yml#L15-L54" rel="nofollow noopener">https://github.com/zopyx/xmldirector.connector/blob/master/.travis.yml#L15-L54</a></p>
<p>This can not be directly expressed as matrix because I want to test only a subset of options that would a matrix give me. So for a number of hand-crafted configurations (expressed through a bunch of environment variables) I want to configure my actions workflow.</p>
<p>How can I do that with Github actions?</p>

**Comment by wipe2238** [comment, marked as answer]
<p>I took your .travis.yml and did a quick edit.</p>
<pre><code>matrix-env:
  runs-on: ubuntu-latest
  strategy:
   matrix:
    env:
    - { TYPE: OWNCLOUD, PLONE_VERSION: 5.2, DOCKER: "pschmitt/owncloud", DOCKER_OPTIONS: "-p 127.0.0.1:8080:80", CONNECTOR_URL: "webdav://localhost:8080/remote.php/webdav", CONNECTOR_USERNAME: "admin", CONNECTOR_PASSWORD: "changeme" }
    - { TYPE: LOCAL_FS, PLONE_VERSION: 5.2, CONNECTOR_URL: "file:///tmp/testing" }
  env: ${{ matrix.env }}
  steps:
  - run: echo "Check Run-&gt;env"
</code></pre>
<p>It should work as you need. Each line inside <code>strategy.matrix.env</code> creates a job with selected env vars.</p>
<p>What is missing is different python versions (which could be passed as yet another variable). I have very little experience with python in Actions so no idea how to select different versions… but (i hope) at least matrix itself won’t be a problem.</p>
